src/core/linalg/src/dense/4C_linalg_tensor.hpp

```diff
@@ -0,0 +1,366 @@
+        requires(storage_type == StorageType::owning)
+      {
+        std::ranges::copy(other.container(), data_.begin());
+      }
+
+      /*!
+       * @brief Access to the underlying raw data of the tensor
+       *
+       * @return Number*
+       */
+      [[nodiscard]] Number* data() { return data_.data(); }
+
+      /*!
+       * @brief Access to the underlying raw data of the tensor for readonly access
+       *
+       * @return Number*
+       */
+      [[nodiscard]] const Number* data() const { return data_.data(); }
+
+      /*!
+       * @brief Access to the underlying container (std::array or std::span, depending on whether it
+       * is owning or a view)
+       *
+       * @return Number*
+       */
+      [[nodiscard]] ContainerType& container() { return data_; }
+
+      /*!
+       * @brief Access to the underlying readonly container (std::array or std::span, depending on
+       * whether it is owning or a view)
+       *
+       * @return Number*
+       */
+      [[nodiscard]] const ContainerType& container() const { return data_; }
+
+      /*!
+       * @brief Indexing operator to access individual values of the tensor (without bound checks)
+       *
+       * @param i
+       * @return Number&
+       */
+      [[nodiscard]] Number& operator()(decltype(n)... i)
+      {
+        return data_[get_flat_index<OrderType::column_major, TensorBoundCheck::no_check, n...>(
+            i...)];
+      }
+
+      /*!
+       * @brief Indexing operator to access individual values of the tensor in readonly mode
+       * (without bound checks)
+       *
+       * @param i
+       * @return Number&
+       */
+      [[nodiscard]] const Number& operator()(decltype(n)... i) const
+      {
+        return data_[get_flat_index<OrderType::column_major, TensorBoundCheck::no_check, n...>(
+            i...)];
+      }
+
+      /*!
+       * @brief Indexing operator to access individual values of the tensor (with bound checks)
+       *
+       * @param i
+       * @return Number&
+       */
+      [[nodiscard]] Number& at(decltype(n)... i)
+      {
+        return data_[get_flat_index<OrderType::column_major, TensorBoundCheck::check, n...>(i...)];
+      }
+
+      /*!
+       * @brief Indexing operator to access individual values of the tensor in readonly mode
+       * (with bound checks)
+       *
+       * @param i
+       * @return Number&
+       */
+      [[nodiscard]] const Number& at(decltype(n)... i) const
+      {
+        return data_[get_flat_index<OrderType::column_major, TensorBoundCheck::check, n...>(i...)];
+      }
+
+      [[nodiscard]] static constexpr std::size_t rank() { return rank_; }
+      [[nodiscard]] static constexpr std::size_t size() { return size_; }
+
+      /*!
+       * @brief Returns the dimension of the i-th axis of the tensor
+       *
+       * @tparam i
+       */
+      template <std::size_t i>
+        requires(i < rank())
+      [[nodiscard]] static constexpr auto size_of()
```

Comment:
```suggestion
      [[nodiscard]] static constexpr auto extent()
```
?

`size_of` is confusing since there is the `sizeof` operator which does something very different.